homeassistant/components/tankerkoenig/__init__.py

```diff
@@ -0,0 +1,203 @@
+
+    def __init__(self, hass, conf):
+        """Initialize the data object."""
+        self._api_key = conf[CONF_API_KEY]
+        self._entities = {}
+        self._fuel_types = conf.get(CONF_FUEL_TYPES)
+
+        latitude = conf.get(CONF_LATITUDE, hass.config.latitude)
+        longitude = conf.get(CONF_LONGITUDE, hass.config.longitude)
+        radius = conf.get(CONF_RADIUS)
+
+        additional_stations = conf.get(CONF_STATIONS)
+
+        _LOGGER.debug("Fetching data for (%s,%s) rad: %s", latitude, longitude, radius)
+
+        try:
+            data = pytankerkoenig.getNearbyStations(
+                self._api_key, latitude, longitude, radius, "all", "dist"
+            )
+        except pytankerkoenig.customException as err:
+            data = {"ok": False, "message": err, "exception": True}
+        _LOGGER.debug("Received data: %s", data)
+        if not data["ok"]:
+            _LOGGER.error(
+                "Error fetching data from tankerkoenig.de:\n%s\nCould not setup sensors",
+                data["message"],
+            )
+            return
+
+        # Add stations found via location + radius
+        nearby_stations = data["stations"]
+        if not nearby_stations:
+            _LOGGER.warning(
+                "Could not find any station in range. Try with a bigger radius"
+            )
+        else:
+            for station in data["stations"]:
+                self.add_station(station)
+
+        # Add manually specified additional stations
+        for station_id in additional_stations:
+            try:
+                additional_station_data = pytankerkoenig.getStationData(
+                    self._api_key, station_id
+                )
+            except pytankerkoenig.customException as err:
+                additional_station_data = {
+                    "ok": False,
+                    "message": err,
+                    "exception": True,
+                }
+
+            if not additional_station_data["ok"]:
+                _LOGGER.error(
+                    "Error when adding station %s:\n %s",
+                    station_id,
+                    additional_station_data["message"],
+                )
+                # Clear entity dictionary, so that the platform setup fails / no sensors get loaded in hass
+                self._entities = {}
+            else:
+                self.add_station(additional_station_data["station"])
+
+    @property
+    def entity_list(self):
+        """Get the list of all entities the platform is monitoring."""
+
+        # Get flat list from the dictionary of lists
+        return [
+            sensor_entity
+            for sublist in self._entities.values()
+            for sensor_entity in sublist
+        ]
+
+    def update(self, now=None):
+        """Get the latest data from tankerkoenig.de."""
+        _LOGGER.debug("Fetching new data from tankerkoenig.de")
+        data = pytankerkoenig.getPriceList(self._api_key, list(self._entities.keys()))
```

Comment:
Extract `list(self._entities.keys()` into a variable before and use it here to make it more readable.